pkg/karmadactl/deinit.go

```diff
@@ -0,0 +1,345 @@
+package karmadactl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+)
+
+// LabelSelector  karmada bootstrapping label
+const LabelSelector = "karmada.io/bootstrapping"
+
+// CommandDeInitOption  options for deinit.
+type CommandDeInitOption struct {
+	Recursive       bool
+	Force           bool
+	KarmadaDataPath string
+	Namespace       string
+	KubeConfig      string
+	KubeClientSet   *kubernetes.Clientset
+}
+
+// NewCmdDeInit remove karmada on kubernetes
+func NewCmdDeInit(cmdOut io.Writer, parentCommand string) *cobra.Command {
+	opts := CommandDeInitOption{}
+	cmd := &cobra.Command{
+		Use:          "deinit",
+		Short:        "remove karmada on kubernetes",
+		Long:         "remove karmada on kubernetes",
+		Example:      deInitExample(parentCommand),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.RunDeInit(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	flags := cmd.PersistentFlags()
+	flags.StringVarP(&opts.Namespace, "namespace", "n", "karmada-system", "Kubernetes namespace")
+	flags.StringVar(&opts.KubeConfig, "kubeconfig", filepath.Join(homeDir(), ".kube", "config"), "absolute path to the kubeconfig file")
```

Comment:
```suggestion
	flags.StringVar(&opts.KubeConfig, "kubeconfig", filepath.Join(homeDir(), ".kube", "config"), "Path to karmada control plane kubeconfig file.")
```

Reply:
I doubt the default `kubeconfig` is useful, as the released binaries is built on `GitHub Action` in a very special home path.

Reply:
This `kubeconfig` seems to be a path to host cluster, not a path to karmada control plane.

Reply:
Oh yes. But the comment for the default value is still valid i guess.